### PR TITLE
82 Refactor DataGettingWrapper & ApiWrapper

### PR DESCRIPTION
**Pull request by OscarUlb**
I would change the names from "initial" to "lastUsed" after merging the branches so we wont run into problems druing the merge.

### CODE REVIEW

**Comment by Kangaroo6942** [comment]
You could already do that since we will have merge conflicts that we need to manually edit anyways, since two parties might change names in the same method for different reasons

File: src/MensaWidget/MWApiWrapper.class.st
Comment (lchristus) [inline src/MensaWidget/MWApiWrapper.class.st:1]: when the class doesn't have variables I don't know we need the header instance vars

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 1]
```diff
@@ -53,19 +53,23 @@ MWDataGettingWrapper class >> canteenDict: anObject [
...
 }
 MWDataGettingWrapper class >> getMealsOnDate: aDate [
 
 	| canteens canteenDict json |
 	
 	canteens := OrderedCollection new.
-	self canteenDict isEmpty ifTrue: [^ canteens add: MWCanteen emptyCanteen;
-		yourself].
+	self canteenDict isEmpty 
+		ifTrue: [^ canteens add: MWCanteen emptyCanteen; yourself].
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:64]: I think we don't need yourself here but not 100% sure

File: src/MensaWidget/MWDataGettingWrapper.class.st
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:1]: remove canteenDict as instance var if not used or documented (edit it shows up as classInstanceVars and I don't quite know what that means tbh )

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 2]
```diff
@@ -53,19 +53,23 @@ MWDataGettingWrapper class >> canteenDict: anObject [
...
 	canteenDict := self canteenDict.
-	canteenDict keys do: [:id |
+	canteens := canteenDict keys collect: [:id |
 		json := MWApiWrapper getMealsFrom: id on: aDate using: WebClient.
-		canteens add: (MWCanteen newWithId: id  Name: (canteenDict at: id) Date: aDate Meals: (MWMealCleaner clean: (MWParser put: json in: MWMeal)))].
+		canteens add: (MWCanteen 
+							newWithId: id  
+							Name: (canteenDict at: id) 
+							Date: aDate 
+							Meals: (MWMealCleaner clean: (MWParser put: json in: MWMeal)))].
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:72]: too many identations?

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 3]
```diff
@@ -161,36 +192,56 @@ MWDataGettingWrapper class >> longitudeMessage [
...
+
+	self initialLatitude: (FillInTheBlankMorph 
+							request: self latitudeMessage 
+							initialAnswer: self initialLatitude);
+		initialLongitude: (FillInTheBlankMorph 
+							request: self longitudeMessage 
+							initialAnswer: self initialLongitude);
+		initialRadius: (FillInTheBlankMorph 
+							request: self radiusMessage 
+							initialAnswer: self initialRadius)
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:216]: also looks like to many indentations

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 4]
```diff
@@ -161,36 +192,56 @@ MWDataGettingWrapper class >> longitudeMessage [
...
 	
-	self initialLatitude: (FillInTheBlankMorph request: self latitudeMessage initialAnswer: self initialLatitude);
-		initialLongitude: (FillInTheBlankMorph request: self longitudeMessage initialAnswer: self initialLongitude);
-		initialRadius: (FillInTheBlankMorph request: self radiusMessage initialAnswer: self initialRadius).
-	outletDict := MWApiWrapper getOutletDictFrom: self initialLatitude and: self initialLongitude andRadius: self initialRadius.
+	self requestUserInput.
+	outletDict := MWApiWrapper 
+								getOutletDictFrom: self initialLatitude 
+								and: self initialLongitude 
+								andRadius: self initialRadius.
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:231]: same here so maybe just looks weird in GitHub

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 5]
```diff
@@ -161,36 +192,56 @@ MWDataGettingWrapper class >> longitudeMessage [
...
+	#'squeak_changestamp' : 'Oscar 7/11/2024 12:57'
 }
 MWDataGettingWrapper class >> switchToDefaultLocation [
 	
 	| outletDict |
 	
-	outletDict := MWApiWrapper getOutletDictFrom: 52.393822 and: 13.129558 andRadius: 5.
+	outletDict := MWApiWrapper getOutletDictFrom: self hpiLatitude 
+									and: self hpiLongitude 
+									andRadius: self hpiRadius.
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapper.class.st:245]: ill add this too but am getting more confident its just Github

File: src/MensaWidget/MWDataGettingWrapperTest.class.st
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapperTest.class.st:1]: you need to update the documentation if the variable doesn't exist anymore

File: src/MensaWidget/MWDataGettingWrapperTest.class.st [hunk 6]
```diff
@@ -27,13 +27,13 @@ MWDataGettingWrapperTest >> testCollectionClasses [
...
 	#category : #tests,
-	#'squeak_changestamp' : 'NK 6/18/2024 21:45'
+	#'squeak_changestamp' : 'Oscar 7/11/2024 13:48'
 }
 MWDataGettingWrapperTest >> testCollectionValues [
 	
 	MWDataGettingWrapper getTodaysMeals do: [:each |
 		each instanceVariableValues do: [:value | 
-			self assert: (value = nil) not]]
+			self assert: (value isNil) not]]
```
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapperTest.class.st:36]: is there not also a isNotNil methode?

File: src/MensaWidget/MWDataGettingWrapperTest.class.st
Comment (lchristus) [inline src/MensaWidget/MWDataGettingWrapperTest.class.st:1]: in test size it is not needed to do new and then class which does the same as accessing the class directly

File: src/MensaWidget/MWFrameTest.class.st
Comment (lchristus) [inline src/MensaWidget/MWFrameTest.class.st:1]: at getAllCurrencies I think the current version would be that the ifTrue would also be in the next line. Or am I wrong @Kangaroo6942?

Comment (lchristus) [reply src/MensaWidget/MWFrameTest.class.st:1]: also in the testRegister analogy I would have the do: in the next line?

**Review by lchristus — CHANGES_REQUESTED** [review]
Very nice some tips here and there but also a lot of it is kind of uncertain since I wasn't able due to current circumstances follow up in squeak directly or we still have some uncertainty in the team. Still I wanted to remark it here so it won't be overlooked :)

File: src/MensaWidget/MWDataGettingWrapper.class.st (hunk 2 above)
Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapper.class.st:72]: It's a trade of. We can make it a wide and short method or a longer one that is easier to read. 

But if you don't like it I can change it back.

File: src/MensaWidget/MWDataGettingWrapper.class.st (hunk 3 above)
Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapper.class.st:216]: Same as the other

File: src/MensaWidget/MWDataGettingWrapper.class.st (hunk 2 above)
Comment (lchristus) [reply src/MensaWidget/MWDataGettingWrapper.class.st:72]: No I like it that way I mean it looks like too many tabs

Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapper.class.st:72]: ahh I see...

File: src/MensaWidget/MWDataGettingWrapperTest.class.st (hunk 6 above)
Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapperTest.class.st:36]: I could not find it.

File: src/MensaWidget/MWDataGettingWrapper.class.st
Comment (Kangaroo6942) [reply src/MensaWidget/MWDataGettingWrapper.class.st:1]: I don't see a canteenDict instantVar...

Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapper.class.st:1]: you are right. I deleted it and it works fine.

File: src/MensaWidget/MWDataGettingWrapperTest.class.st (hunk 6 above)
Comment (lchristus) [reply src/MensaWidget/MWDataGettingWrapperTest.class.st:30]: should be notNil

Comment (OscarUlb) [reply src/MensaWidget/MWDataGettingWrapperTest.class.st:30]: got it

**Review by lchristus — APPROVED** [review]
very nice nothing more that I could find

**Pull request merged**